transforms/StickbreakingPowerLogistic/StickbreakingPowerLogistic_functions.stan

```diff
@@ -0,0 +1,36 @@
+vector stickbreaking_power_logistic_simplex_constrain_lp(vector y) {
+  int N = rows(y) + 1;
+  vector[N] x;
+  real log_u, log_w, log_z;
+  real log_cum_prod = 0;
+  for (i in 1 : (N - 1)) {
+    log_u = log_inv_logit(y[i]); // logistic_lcdf(y[i] | 0, 1);
+    log_w = log_u / (N - i);
+    log_z = log1m_exp(log_w);
+    x[i] = exp(log_cum_prod + log_z);
+    target += 2 * log_u - y[i]; // logistic_lupdf(y[i] | 0, 1);
+    log_cum_prod += log1m_exp(log_z);
+  }
+  x[N] = exp(log_cum_prod);
+  target += -lgamma(N);
+  return x;
+}
+
+vector stickbreaking_power_logistic_log_simplex_constrain_lp(vector y) {
+  int N = rows(y) + 1;
+  vector[N] log_x;
+  real log_u, log_w, log_z;
+  real log_cum_prod = 0;
+  for (i in 1 : (N - 1)) {
+    log_u = log_inv_logit(y[i]); // logistic_lcdf(y[i] | 0, 1);
+    log_w = log_u / (N - i);
+    log_z = log1m_exp(log_w);
+    log_x[i] = log_cum_prod + log_z;
+    target += 2 * log_u - y[i]; // logistic_lupdf(y[i] | 0, 1);
+    target += -log_x[i];
```

Comment:
It's more efficient to pull this out of the loop and just use the vectorized `target +=`:

```stan
target += -log_x;
```

Reply:
Ah, okay, so `-log_x` doesn't end up allocating a new vector, or that doesn't really matter?
What about if I need to add the contents of several vectors `a` and `b` of the same length to the target. Is it better to do `target += a; target += b` or `target += a + b`, or are both the same?

Reply:
Actually, `-log_x` might allocate a new vector unless we're returning an expression graph.  Steve Bronder is going through and replacing the return types with expression templates, so that things like `-log_x` will not allocate a new vector.  When that's all done, vectorization will pretty much always be the right thing to do.

One issue with optimization is that as the compiler changes, so do the optimization techniques.  For example, ll the careful work I'd done on Java 1.4 optimization in the early 00s went out the window when they built most of it into the Java 1.5 compiler.  

I answered the `target +=` question above---better to split it.